File: test/language/templates/java/templates/InstantiateSimpleTest.java
package templates;

import static org.junit.jupiter.api.Assertions.*;
import org.junit.jupiter.api.Test;

import java.io.IOException;

import zserio.runtime.io.BitStreamReader;
import zserio.runtime.io.ByteArrayBitStreamReader;
import zserio.runtime.io.ByteArrayBitStreamWriter;

import templates.instantiate_simple.InstantiateSimple;
import templates.instantiate_simple.U32;

public class InstantiateSimpleTest
{
    @Test
    public void readWrite() throws IOException
    {
        final InstantiateSimple instantiateSimple = new InstantiateSimple();
        instantiateSimple.setTest(new U32(13));

        final ByteArrayBitStreamWriter writer = new ByteArrayBitStreamWriter();
        instantiateSimple.write(writer);

        final BitStreamReader reader = new ByteArrayBitStreamReader(
                writer.toByteArray(), writer.getBitPosition());
        final InstantiateSimple readInstantiateSimple = new InstantiateSimple(reader);
        assertTrue(instantiateSimple.equals(readInstantiateSimple));
    }
}

File: test/extensions/language/templates/cpp/InstantiateSimpleTest.cpp
#include "gtest/gtest.h"
#include "templates/instantiate_simple/InstantiateSimple.h"

namespace templates
{
namespace instantiate_simple
{

TEST(InstantiateSimpleTest, readWrite)
{
    InstantiateSimple instantiateSimple;
    instantiateSimple.setTest(U32{13});

    zserio::BitBuffer bitBuffer = zserio::BitBuffer(1024 * 8);
    zserio::BitStreamWriter writer(bitBuffer);
    instantiateSimple.write(writer);

    zserio::BitStreamReader reader(writer.getWriteBuffer(), writer.getBitPosition(), zserio::BitsTag());
    InstantiateSimple readInstantiateSimple(reader);

    ASSERT_TRUE(instantiateSimple == readInstantiateSimple);
}

} // namespace instantiate_simple
} // namespace templates
